Week-1/assignment1_1.md

```diff
@@ -0,0 +1,107 @@
+
+#How The Browser Work
+
+
+When a user enters an URL in the browser, how does the browser fetch the desiredresult ? 
+
+when A enter url(url stands for universal resource locator) 
+ 
+now the browser need to know which server i need to connect.This is done with DNS(Phone book of internet) DNS translates domain names to IP addresses so browser can load resources .To make look up  fast,The DNS information is heavily cached.
+first browser itself caches it for a short period of time
+if not in browser and browser ask for operating system(os also cached dns information for some period)
+
+
+If OS also dont have ,Browser looks up and makes a query out to internet  to a DNS resolver.This is set off a chain of requests untill the IP address is resolved
+
+
+Now the browser has IP addresss
+
+browser establish a TCP connection with the server  Web server
+
+browser send HTTp request to server
+
+server sends back HTTP reponse
```

Comment:
server send back http response with some http codes like 200, 404, etc mention those and read about http codes

Reply:
I will learn mam